webapp/src/components/sidebar/deleteBoardDialog.tsx

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React, {useState} from 'react'
+import {FormattedMessage} from 'react-intl'
+
+import {Utils} from '../../utils'
+import Button from '../../widgets/buttons/button'
+
+import Dialog from '../dialog'
+import RootPortal from '../rootPortal'
+
+import './deleteBoardDialog.scss'
+
+type Props = {
+    boardTitle: string;
+    onClose: () => void;
+    onDelete: () => Promise<void>
+}
+
+export default function DeleteBoardDialog(props: Props): JSX.Element {
+    const [isSubmitting, setSubmitting] = useState(false)
+
+    return (
+        <RootPortal>
+            <Dialog
+                onClose={close}
+                toolsMenu={null}
+                className='DeleteBoardDialog'
+            >
+                <div className='container'>
+                    <h2 className='header'>
+                        <FormattedMessage
+                            id='DeleteBoardDialog.confirm-tite'
+                            defaultMessage='Confirm Delete Board'
+                        />
+                    </h2>
+                    <p className='body'>
+                        <FormattedMessage
+                            id='DeleteBoardDialog.confirm-info'
+                            defaultMessage='Are you sure you want to delete the board “{boardTitle}”? Deleting it will delete the property from all cards in this board.'
+                            values={{
+                                boardTitle: props.boardTitle,
+                            }}
+                        />
+                    </p>
+                    <div className='footer'>
+                        <Button onClick={() => !isSubmitting && props.onClose()}>
+                            <FormattedMessage
+                                id='DeleteBoardDialog.confirm-cancel'
+                                defaultMessage='Cancel'
+                            />
+                        </Button>
+                        <Button
+                            filled={true}
+                            danger={true}
+                            onClick={async () => {
+                                try {
+                                    setSubmitting(true)
+                                    await props.onDelete()
+                                    setSubmitting(false)
+                                    props.onClose()
+                                } catch (e) {
+                                    setSubmitting(false)
+                                    Utils.logError(`Delete board ERROR: ${e}`)
+
+                                    // TODO: display error on screen
```

Comment:
Do we want to address this?

Reply:
I'm not sure, I could not find existing component for displaying toast messages, or errors in general. If there is such implementation already existing, then I would use it, otherwise IMO it should addressed separately.

Reply:
@asaadmahmood Would you be able to add a variant to [this figma file](https://www.figma.com/file/a3SJgKolniXEU2G4zG8Uc0/Issue-512-Deleting-a-property?node-id=214%3A3130) for situation when delete operation causes an unexpected error?